ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -397,14 +413,25 @@ public void onCompletion(String result, Exception exception) {
           logger.trace("Forwarding {} of {} to the router", restMethod, result);
           switch (restMethod) {
             case GET:
-              getCallback.markStartTime();
-              router.getBlob(result, getCallback.options, getCallback);
+              RestUtils.SubResource subresource = RestUtils.getBlobSubResource(restRequest);
+              if (subresource == null || subresource.equals(RestUtils.SubResource.BlobInfo) || subresource.equals(
```

Comment:
This would make it easier to inspect that we are handling all subresource cases if more are added in the future